File: 2017/02/Python/checksum.py
#!/usr/bin/env python3

from itertools import combinations as cmbs

print(sum( [max(l) -  min(l) for l in [ [int(x) for x in line.split('\t')] for line in open("input")] ] ))

print(sum( [max(t) // min(t) for l in [ [int(x) for x in line.split('\t')] for line in open("input")] for t in cmbs(l, 2) if not max(t) % min(t) ]))

##############################
# Author: Rahul Butani       #
# Date:   December 2nd, 2017 #
##############################

File: 2017/02/Haskell/checksum.hs
#!/usr/bin/env runhaskell

import Data.List

main = do
    g <- readFile("input")
    let l = [ [ read w :: Int | w <- h] | l <- lines g, let h = words l]
    
    print( sum( [ maximum x - minimum x | x <- l ] ) )
    print( sum (concat [ [ maximum x `div` minimum x | x <- r, maximum x `mod` minimum x == 0 ] | r <- [ [ x:y:[] | (x:xs) <- tails ls, y:_ <- tails xs ] | ls <- l]]))

--------------------------------
-- Author: Rahul Butani       --
-- Date:   December 2nd, 2017 --
--------------------------------
